active/0002-packager.md

```diff
@@ -0,0 +1,257 @@
+
+```
+ 'browserified/ember-qunit/ember-qunit-legacy.js',
+ 'dummy-tests/dep-graph.json',
+ 'dummy-tests/index.html',
+ 'dummy-tests/index.js',
+ 'dummy-tests/unit/components/foo-bar-test.js',
+ 'dummy/app.js',
+ 'dummy/components/baz-bar.js',
+ 'dummy/components/foo-bar.js',
+ 'dummy/config/environment.js',
+ 'dummy/crossdomain.xml',
+ 'dummy/dep-graph.json',
+ 'dummy/index.html',
+ 'dummy/index.js',
+ 'dummy/pods/bizz-buzz/component.js',
+ 'dummy/pods/bizz-buzz/template.js',
+ 'dummy/pods/foo-baz/component.js',
+ 'dummy/pods/foo-baz/template.js',
+ 'dummy/robots.txt',
+ 'dummy/router.js',
+ 'dummy/styles/app.css',
+ 'dummy/templates/components/foo-bar.js',
+ 'dummy/templates/profile.js',
+ 'ember-qunit/dep-graph.json',
+ 'ember-qunit/ember-qunit.js',
+ 'ember/dep-graph.json',
+ 'ember/ember.js',
+ 'ember/ember/get.js'
+```
+
+At this point we are ready to start concatenating the files together.  Out of the box the resolver will have 4 concatenation strategies: 'default', 'engines', 'http2', and 'custom'.
+
+Since the graph has already been resolved you can provide an array of strategies and the output will result in the dist containing a folder for each strategy.
+
+## Strategy Definitions
+
+- __default__: This is the same output that we have today; app and vendor for both js and css, along with index.html, crossdomain.xml and robots.txt. This may eventually get re-named to legacy.
+- __engines__: Once we have the concept of engines, developers will more than likely want to break apart into functional sets. This will allow you to build a bundle for each engine. These are looked at as entries to the pre-packager.
+- __http2__: As more servers move towards using [HTTP2](https://http2.github.io/) you actually want granular assets instead of concated blobs. This is due to the fact that you have [multiplexing](https://www.mnot.net/blog/2014/01/30/http2_expectations) in HTTP2. This allows the browser to cache bust only on the things that have changed. This may change to "none".
+- __custom__: If none of the out of the box steps work for you can tell the packager that you're are going to write your own concatenation logic.
+
+### Custom Concatenation
+As part of the options you can set the `composeOutput` property to a function that will receive the tree and the resolved dependency graph. It's then up to you to preform the concatenation and return the the tree.
+
+### Post Build Addon Hooks
+Once the tree has been concated, we then send that tree through the addon post-build hooks. This completes the the addon hook lifecycle.
+
+# Migration Plan
+
+## Ember CLI Developers
+
+We need to move all of the client dependencies that Ember CLI relies on to start publishing to npm as addons and are resolve their dependencies within node_modules as oppossed to bower_components. We will also need to rename some of the modules as they are at unreachable code paths e.g. `ember/resolver` does not actually resolve to the ember namespace.
```

Comment:
we need a bower compatibility layer, because people are using bower for many things not controlled by ember-cli. That being said, `app.import` will likely just "make it work" for now.